src/node_crypto_clienthello.cc

```diff
@@ -146,7 +149,8 @@ void ClientHelloParser::ParseExtension(const uint16_t type,
       ocsp_request_ = 1;
       break;
     case kTLSSessionTicket:
-      tls_ticket_size_ = len;
+      // TBD POSSIBLE DATA LOSS:
+      tls_ticket_size_ = static_cast<uint16_t>(len & kTLSTicketSizeMask);
```

Comment:
Here is what I gather from a quick grep: `tls_ticket_size_` stores a ticket size and `tls_ticket_` stores a pointer to some ticket information but they are only used to determine a `hello.has_ticket_` boolean value. I start to wonder if we could replace the `tls_ticket_` and `tls_ticket_size_` members with a member like `has_valid_tls_ticker_` to avoid keeping more state than necessary?